### ISSUE TITLE
Allow using system version of JUCE

### ISSUE THREAD

**Issue by dvzrv** [monocasual/giada#526]
**Is your feature request related to a problem? Please describe.**
JUCE is shipped as a system package on many downstream distributions.
As it offers cmake integration, it can be found and used like any other dependency.

**Describe the solution you'd like**
Either automatically detect system wide JUCE or optionally offer to detect it on request (e.g. using a cmake option such as `USE_SYSTEM_JUCE`, that defaults to `OFF`).

**Describe alternatives you've considered**
Patching everything is painful.

**Additional context**
Upstream JUCE has added proper cmake integration for all kinds of things and it might be way easier directly using their stuff.
E.g. stochas uses functions that JUCE exposes via their cmake integration: https://github.com/surge-synthesizer/stochas/blob/87a8e064d71e49140cddbc74b62b573a9fd06e1e/CMakeLists.txt

**Comment by gvnnz** [comment]
Good point. Currently we are using JUCE in a pretty unorthodox way, that is: modules are included "by hand" (see [here](https://github.com/monocasual/giada/blob/master/CMakeLists.txt#L495)) without using the helper functions provided by JUCE and its CMake integration.

**Comment by gvnnz** [comment]
We have been using JUCE's CMake integration for a while now, so adding the ability to use the system version should be easy. However I'm not 100% sure how to fix the [add_subdirectory](https://github.com/monocasual/giada/blob/958c13f24ec08f97108be282d66eaa2d17e2d4b1/CMakeLists.txt#L564) step. @dvzrv got any clues?

**Comment by grawlinson** [comment]
Here's a rough idea that could work:

```
if(USE_SYSTEM_JUCE)
  find_package(JUCE CONFIG REQUIRED)
else()
  add_subdirectory(${CMAKE_SOURCE_DIR}/src/deps/juce)
endif()
```

I don't quite know enough about CMake conditionals, so it may need some tweaking.
